src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -239,12 +249,19 @@ private static ParameterDescriptor CreateParameterDescriptor(ParameterInfo param
         {
             return (BindingSource.Body, parameter.Name ?? string.Empty, fromBodyAttribute.AllowEmpty, parameter.ParameterType);
         }
+        else if (attributes.OfType<IFromFormMetadata>().FirstOrDefault() is { } fromFormAttribute)
+        {
+            // TODO Should this be Form or FormFile if parameter.ParameterType is IFormFileCollection?
+            // Currently it's Services (see below) - which is the more appropriate?
```

Comment:
I think we should probably use `BindingSource.FormFile` for both `IFormFile` and `IFormFileCollection`. 

@domaindrivendev @RicoSuter @captainsafia

Reply:
https://github.com/dotnet/aspnetcore/blob/a450cb69b5e4549f5515cdb057a68771f56cefd7/src/Mvc/Mvc.Core/src/Infrastructure/MvcCoreMvcOptionsSetup.cs#L131-L134